pyEpiabm/pyEpiabm/tests/test_unit/test_routine/test_simulation.py

```diff
@@ -232,18 +348,109 @@ def test_s_write_to_file(self, mock_mkdir):
-            data["time"] = time
-            cell = self.test_population.cells[0]
-            data["cell"] = cell.id
-            data["location_x"] = cell.location[0]
-            data["location_y"] = cell.location[0]
+            for cell in self.test_population.cells:
+                for age_i in \
+                        range(0,
+                              len(pe.Parameters.instance().age_proportions)):
+                    for inf_status in data:
+                        data_per_inf_status = \
+                                cell.compartment_counter.retrieve()[inf_status]
+                        data[inf_status] += data_per_inf_status[age_i]
+                data["age_group"] = \
+                    len(pe.Parameters.instance().age_proportions)
+                data["time"] = time
+                cell = self.test_population.cells[0]
+                data["cell"] = cell.id
+                data["location_x"] = cell.location[0]
+                data["location_y"] = cell.location[0]
 
             with patch.object(spatial_sim.writer, 'write') as mock:
                 spatial_sim.write_to_file(time)
                 mock.assert_called_with(data)
         mock_mkdir.assert_called_with(os.path.join(os.getcwd(),
-                                      self.file_params["output_dir"]))
+                                                   self.file_params[
+                                                       "output_dir"]))
+
+    @patch('os.makedirs')
+    def test_write_to_ih_file(self, mock_mkdir):
+
+        if os.path.exists(self.ih_file_params["output_dir"]):
+            os.rmdir(self.ih_file_params["output_dir"])
+
+        mo = mock_open()
+        self.ih_file_params['status_output'] = True
+        self.ih_file_params['infectiousness_output'] = False
+        with patch('pyEpiabm.output._csv_dict_writer.open', mo):
+            time = 1
+            test_sim = pe.routine.Simulation()
+            test_sim.configure(self.test_population, self.initial_sweeps,
+                               self.sweeps, self.sim_params, self.file_params,
+                               self.ih_file_params)
+            data = {column: 0 for column in
+                    test_sim.ih_status_writer.writer.fieldnames}
+            for cell in self.test_population.cells:
+                for person in cell.persons:
+                    data[person.id] = person.infection_status.value
+            data["time"] = time
+
+            with patch.object(test_sim.ih_status_writer, 'write') as mock:
+                test_sim.write_to_ih_file(time, "status")
+                mock.assert_called_with(data)
+        mock_mkdir.assert_called_with(os.path.join(os.getcwd(),
+                                      self.ih_file_params["output_dir"]))
+
+        self.ih_file_params['status_output'] = False
+        self.ih_file_params['infectiousness_output'] = True
+        with patch('pyEpiabm.output._csv_dict_writer.open', mo):
+            time = 1
```

Comment:
Perhaps make this a argument (with default value 1) to the original test function